script/balances.ts

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env bun
+
+import { defineCommand, runMain } from 'citty'
+import { consola } from 'consola'
+import type { TronWeb } from 'tronweb'
+import {
+  createPublicClient,
+  http,
+  formatEther,
+  isAddress as isValidEvmAddress,
+} from 'viem'
+
+import networksConfig from '../config/networks.json'
+
+import { initTronWeb } from './troncast/utils/tronweb'
+import { node_url } from './utils/network'
+
+// Load networks configuration
+
+interface INetworkConfig {
+  name: string
+  chainId: number
+  nativeAddress: string
+  nativeCurrency: string
+  wrappedNativeAddress: string
+  status: string
+  type: string
+  rpcUrl: string
+  verificationType: string
+  explorerUrl: string
+  explorerApiUrl: string
+  multicallAddress: string
+  safeAddress: string
+  gasZipChainId: number
+  isZkEVM: boolean
+  deployedWithEvmVersion: string
+  deployedWithSolcVersion: string
+  create3Factory?: string
+  devNotes?: string
+}
+
+interface IBalanceResult {
+  network: string
+  chainId: number
+  nativeCurrency: string
+  balance: string
+  formattedBalance: string
+  error?: string
+}
+
+interface ITableRow {
+  Network: string
+  'Chain ID': number
+  Currency: string
+  Balance: string
+  Status: string
+}
+
+async function getTronBalance(
+  tronWeb: TronWeb,
+  address: string,
+  network: INetworkConfig
+): Promise<IBalanceResult> {
+  try {
+    // Convert EVM address to Tron address if needed
+    const tronAddress = address.startsWith('0x')
+      ? tronWeb.address.fromHex(address)
+      : address
+
+    // Get TRX balance (in SUN, 1 TRX = 1,000,000 SUN)
+    const balanceInSun = await tronWeb.trx.getBalance(tronAddress)
+    const balanceInTrx = balanceInSun / 1_000_000
+
+    return {
+      network: network.name,
+      chainId: network.chainId,
+      nativeCurrency: network.nativeCurrency,
+      balance: balanceInSun.toString(),
+      formattedBalance: balanceInTrx.toFixed(6),
+    }
+  } catch (error) {
+    return {
+      network: network.name,
+      chainId: network.chainId,
+      nativeCurrency: network.nativeCurrency,
+      balance: '0',
+      formattedBalance: '0',
+      error: error instanceof Error ? error.message : 'Unknown error',
+    }
+  }
+}
+
+async function getEvmBalance(
+  network: INetworkConfig,
+  address: string
+): Promise<IBalanceResult> {
+  try {
+    // Try to get RPC URL from environment variable first, fallback to networks.json
+    const rpcUrl = node_url(network.name) || network.rpcUrl
+
+    const client = createPublicClient({
+      transport: http(rpcUrl, {
+        timeout: 10_000,
+        retryCount: 2,
+        retryDelay: 1000,
```

Comment:
interesting...should we add this to our other helper functions, too?

Reply:
Added a ticket https://lifi.atlassian.net/jira/software/c/projects/LF/boards/5?selectedIssue=LF-15417

Reply:
awesome, thanks. 
You can resolve such comments directly btw